### ISSUE TITLE
I18n::InvalidLocale error when processing happens to fail and :en is not a valid locale

### ISSUE THREAD

**Issue by attilagyorffy** [carrierwaveuploader/carrierwave#1902]
Hi,

This is an odd one that happens to break my app. I know that ideally the processing (using MiniMagic btw) should not break but in some cases it may actually do and I don't think my app should with the following exception:

``` ruby
I18n::InvalidLocale
```

This happens to be an issue where the MiniMagic processing plugin tries to log an issue and happens to be using the I18n translations with an explicit `:en` locale. Trouble is: :en is not an available locale in my app and so i18n starts to complain (for a valid reason).

Here's what's happening:

```
I18n::InvalidLocale (:en is not a valid locale):
  i18n (0.7.0) lib/i18n.rb:284:in `enforce_available_locales!'
  i18n (0.7.0) lib/i18n.rb:151:in `translate'
  carrierwave (0.10.0) lib/carrierwave/processing/mini_magick.rb:271:in `rescue in manipulate!'
  carrierwave (0.10.0) lib/carrierwave/processing/mini_magick.rb:259:in `manipulate!'
  carrierwave (0.10.0) lib/carrierwave/processing/mini_magick.rb:131:in `resize_to_limit'
  carrierwave (0.10.0) lib/carrierwave/uploader/processing.rb:84:in `block in process!'
  carrierwave (0.10.0) lib/carrierwave/uploader/processing.rb:76:in `each'
  carrierwave (0.10.0) lib/carrierwave/uploader/processing.rb:76:in `process!'
  carrierwave (0.10.0) lib/carrierwave/uploader/callbacks.rb:18:in `block in with_callbacks'
  carrierwave (0.10.0) lib/carrierwave/uploader/callbacks.rb:18:in `each'
  carrierwave (0.10.0) lib/carrierwave/uploader/callbacks.rb:18:in `with_callbacks'
  carrierwave (0.10.0) lib/carrierwave/uploader/cache.rb:122:in `cache!'
  carrierwave (0.10.0) lib/carrierwave/uploader/versions.rb:274:in `block in cache_versions!'
  carrierwave (0.10.0) lib/carrierwave/uploader/versions.rb:259:in `each'
  carrierwave (0.10.0) lib/carrierwave/uploader/versions.rb:259:in `cache_versions!'
  carrierwave (0.10.0) lib/carrierwave/uploader/callbacks.rb:18:in `block in with_callbacks'
  carrierwave (0.10.0) lib/carrierwave/uploader/callbacks.rb:18:in `each'
  carrierwave (0.10.0) lib/carrierwave/uploader/callbacks.rb:18:in `with_callbacks'
  carrierwave (0.10.0) lib/carrierwave/uploader/cache.rb:122:in `cache!'
  carrierwave (0.10.0) lib/carrierwave/mount.rb:329:in `cache'
  carrierwave (0.10.0) lib/carrierwave/mount.rb:163:in `picture='
  carrierwave (0.10.0) lib/carrierwave/orm/activerecord.rb:39:in `picture='
  activerecord (4.2.5.1) lib/active_record/attribute_assignment.rb:54:in `public_send'
  activerecord (4.2.5.1) lib/active_record/attribute_assignment.rb:54:in `_assign_attribute'
  activerecord (4.2.5.1) lib/active_record/attribute_assignment.rb:41:in `block in assign_attributes'
  actionpack (4.2.5.1) lib/action_controller/metal/strong_parameters.rb:185:in `each_pair'
  actionpack (4.2.5.1) lib/action_controller/metal/strong_parameters.rb:185:in `each_pair'
...
```

The relevant line in the code:

https://github.com/carrierwaveuploader/carrierwave/blob/master/lib/carrierwave/processing/mini_magick.rb#L306

I just assume this is because someone wanted to have a single source of truth and wanted to keep the error message in a single place (the localisations).

Can we use the current locale in case this has any meaning to an end user OR leave the translation out of this completely in case this is just for logging?

Thanks in advance.

**Comment by liquid** [comment]
Now that I've seen CarrierWave is advertised as a solution that is supposed to be compatible with Sinatra, I am more confident that I18n should not be an explicit dependency of the library. (I realise it is not in the gemspec, but essentially the line above makes use of `I18n` anyway.

**Comment by mreq** [comment]
Happens in rails 4.2.6 as well.

**Comment by mshibuya** [comment]
Dup of #1729, and fixed by #2083.
